Windows: build broken again

```
error[E0277]: `std::ffi::OsStr` doesn't implement `std::fmt::Display`
  --> C:\Users\Ron\.cargo\registry\src\github.com-1ecc6299db9ec823\jokeyrhyme-dotfiles-0.25.0\src\lib\goget.rs:34:75
   |
34 |                     let bin = gopath().join("bin").join(format!("{}.exe", f));
   |                                                                           ^ `std::ffi::OsStr` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `std::ffi::OsStr`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `std::fmt::Display` for `&std::ffi::OsStr`
   = note: required by `std::fmt::Display::fmt`

error[E0277]: `std::ffi::OsStr` doesn't implement `std::fmt::Display`
  --> C:\Users\Ron\.cargo\registry\src\github.com-1ecc6299db9ec823\jokeyrhyme-dotfiles-0.25.0\src\lib\goget.rs:85:67
   |
85 |             let bin = gopath().join("bin").join(format!("{}.exe", f));
   |                                                                   ^ `std::ffi::OsStr` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `std::ffi::OsStr`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `std::fmt::Display` for `&std::ffi::OsStr`
   = note: required by `std::fmt::Display::fmt`

error: aborting due to 2 previous errors
```

- probably best to avoid target-specific code except where there's a significant advantage in keeping it out of builds for other targets

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Changed
+
+- check OS at runtime rather than at compile time in more places
+
+### Fixed
+
+- Windows: builds again (probably), yay
+
 ## [0.25.0] - 2018-12-02
 
 ### Added

--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ lint:
 	# rustup target add x86_64-unknown-linux-gnu || true
 	# cargo check --all-features --target x86_64-unknown-linux-gnu
 	cargo check --all-features --benches --tests
-	cargo clippy -- --deny clippy::all
+	cargo clippy --all-features --benches --tests -- --deny clippy::all
 
 .NOCACHE: test
 test:

--- a/src/lib/cargo.rs
+++ b/src/lib/cargo.rs
@@ -1,4 +1,4 @@
-use std::{collections::HashMap, io, path::PathBuf, str};
+use std::{collections::HashMap, env::consts::OS, io, path::PathBuf, str};
 
 use regex;
 
@@ -82,12 +82,11 @@ where
 }
 
 fn cargo_exe() -> PathBuf {
-    #[cfg(windows)]
-    let pb = bin_dir().join("cargo.exe");
-    #[cfg(not(windows))]
-    let pb = bin_dir().join("cargo");
-
-    pb
+    bin_dir().join(if OS == "windows" {
+        "cargo.exe"
+    } else {
+        "cargo"
+    })
 }
 
 #[cfg_attr(feature = "cargo-clippy", allow(needless_pass_by_value))]

--- a/src/lib/env.rs
+++ b/src/lib/env.rs
@@ -1,4 +1,8 @@
-use std::{env::var, fmt::Display, path::PathBuf};
+use std::{
+    env::{consts::OS, var},
+    fmt::Display,
+    path::PathBuf,
+};
 
 pub const EDITOR: &str = "EDITOR";
 pub const PATH: &str = "PATH";
@@ -34,10 +38,13 @@ impl Exports {
             .into_iter()
             .map(|p| p.as_path().to_string_lossy().into_owned())
             .collect();
-        #[cfg(windows)]
-        let path_line = export_shell(&shell, PATH, path_strings.join(";").as_str());
-        #[cfg(not(windows))]
-        let path_line = export_shell(&shell, PATH, path_strings.join(":").as_str());
+        let path_line = export_shell(
+            &shell,
+            PATH,
+            path_strings
+                .join(if OS == "windows" { ";" } else { ":" })
+                .as_str(),
+        );
 
         let lines: &[String] = &[editor_line, path_line];
         lines.join("\n")
@@ -49,11 +56,7 @@ impl Default for Exports {
         let mut exports = Self::new();
         exports.path = match var("PATH") {
             Ok(paths) => {
-                #[cfg(windows)]
-                let path_strings = paths.split(";");
-                #[cfg(not(windows))]
-                let path_strings = paths.split(":");
-
+                let path_strings = paths.split(if OS == "windows" { ";" } else { ":" });
                 path_strings.map(|p| PathBuf::from(p)).collect()
             }
             Err(_) => exports.path,

--- a/src/lib/goget.rs
+++ b/src/lib/goget.rs
@@ -1,4 +1,4 @@
-use std::{io, path::PathBuf};
+use std::{env::consts::EXE_SUFFIX, io, path::PathBuf};
 
 use crate::{
     lib::favourites::Favourites,
@@ -30,11 +30,10 @@ impl Favourites for GoGetFavourites {
 
             match p.file_name() {
                 Some(f) => {
-                    #[cfg(windows)]
-                    let bin = gopath().join("bin").join(format!("{}.exe", f));
-                    #[cfg(not(windows))]
-                    let bin = gopath().join("bin").join(f);
-
+                    let bin =
+                        gopath()
+                            .join("bin")
+                            .join(format!("{}{}", f.to_string_lossy(), EXE_SUFFIX));
                     delete_if_exists(bin);
                 }
                 None => {}
@@ -81,10 +80,9 @@ where
     let p = PathBuf::from(pkg.as_ref());
     match p.file_name() {
         Some(f) => {
-            #[cfg(windows)]
-            let bin = gopath().join("bin").join(format!("{}.exe", f));
-            #[cfg(not(windows))]
-            let bin = gopath().join("bin").join(f);
+            let bin = gopath()
+                .join("bin")
+                .join(format!("{}{}", f.to_string_lossy(), EXE_SUFFIX));
 
             let src = gopath().join("src").join(&p);
             bin.is_file() && src.is_dir()
@@ -95,6 +93,8 @@ where
 
 #[cfg(test)]
 mod tests {
+    use std::env::consts::OS;
+
     use super::*;
 
     #[test]
@@ -115,10 +115,11 @@ mod tests {
             uninstall: vec![],
         };
 
-        #[cfg(windows)]
-        let bin = gopath().join("bin").join("goimports.exe");
-        #[cfg(not(windows))]
-        let bin = gopath().join("bin").join("goimports");
+        let bin = gopath().join("bin").join(if OS == "windows" {
+            "goimports.exe"
+        } else {
+            "goimports"
+        });
 
         let src = gopath().join("src").join(goimports_pkg.clone());
         assert_eq!(

--- a/src/lib/rust.rs
+++ b/src/lib/rust.rs
@@ -1,4 +1,4 @@
-use std::{io, path::PathBuf};
+use std::{env::consts::OS, io, path::PathBuf};
 
 use crate::utils::{
     env::home_dir,
@@ -45,10 +45,9 @@ pub fn rustup_version() -> String {
 }
 
 fn rustup_exe() -> PathBuf {
-    #[cfg(windows)]
-    let pb = bin_dir().join("rustup.exe");
-    #[cfg(not(windows))]
-    let pb = bin_dir().join("rustup");
-
-    pb
+    bin_dir().join(if OS == "windows" {
+        "rustup.exe"
+    } else {
+        "rustup"
+    })
 }

--- a/src/tasks/dep.rs
+++ b/src/tasks/dep.rs
@@ -1,3 +1,5 @@
+use std::env::consts::OS;
+
 use crate::lib::{
     ghrtask::GHRTask,
     task::{self, Task},
@@ -25,10 +27,11 @@ const GHR_TASK: GHRTask = GHRTask {
 };
 
 fn asset_filter(asset: &Asset) -> bool {
-    #[cfg(windows)]
-    let name = format!("dep-{}-{}.exe", os(), arch());
-    #[cfg(not(windows))]
-    let name = format!("dep-{}-{}", os(), arch());
+    let name = if OS == "windows" {
+        format!("dep-{}-{}.exe", os(), arch())
+    } else {
+        format!("dep-{}-{}", os(), arch())
+    };
 
     asset.name == name
 }

--- a/src/tasks/gitleaks.rs
+++ b/src/tasks/gitleaks.rs
@@ -1,9 +1,15 @@
-use crate::lib::{
-    ghrtask::GHRTask,
-    task::{self, Task},
+use std::env::consts::EXE_SUFFIX;
+
+use crate::{
+    lib::{
+        ghrtask::GHRTask,
+        task::{self, Task},
+    },
+    utils::{
+        github::Asset,
+        golang::{arch, os},
+    },
 };
-use crate::utils::github::Asset;
-use crate::utils::golang::{arch, os};
 
 pub fn task() -> Task {
     Task {
@@ -25,11 +31,7 @@ const GHR_TASK: GHRTask = GHRTask {
 };
 
 fn asset_filter(asset: &Asset) -> bool {
-    #[cfg(windows)]
-    let name = format!("gitleaks-{}-{}.exe", os(), arch());
-    #[cfg(not(windows))]
-    let name = format!("gitleaks-{}-{}", os(), arch());
-
+    let name = format!("gitleaks-{}-{}{}", os(), arch(), EXE_SUFFIX);
     asset.name == name
 }
 

--- a/src/tasks/golang.rs
+++ b/src/tasks/golang.rs
@@ -1,3 +1,5 @@
+use std::env::consts::OS;
+
 use crate::lib::{
     env::Exports,
     task::{self, Status, Task},
@@ -39,31 +41,24 @@ where
 
     let temp_path = mktemp()?;
 
-    #[cfg(windows)]
     let remote_url = format!(
-        "https://dl.google.com/go/{}.{}-{}.zip",
+        "https://dl.google.com/go/{}.{}-{}.{}",
         version.as_ref(),
         os(),
-        arch()
+        arch(),
+        if OS == "windows" { "exe" } else { "tar.gz" },
     );
-    #[cfg(not(windows))]
-    let remote_url = format!(
-        "https://dl.google.com/go/{}.{}-{}.tar.gz",
-        version.as_ref(),
-        os(),
-        arch()
-    );
-
     utils::http::download(remote_url, &temp_path)?;
 
     let local_path = utils::env::home_dir().join(".local");
     let target_path = local_path.join("go");
     utils::fs::delete_if_exists(&target_path);
 
-    #[cfg(windows)]
-    utils::archive::extract_zip(&temp_path, &local_path)?;
-    #[cfg(not(windows))]
-    utils::archive::extract_tar_gz(&temp_path, &local_path)?;
+    if OS == "windows" {
+        utils::archive::extract_zip(&temp_path, &local_path)?;
+    } else {
+        utils::archive::extract_tar_gz(&temp_path, &local_path)?;
+    }
 
     utils::fs::delete_if_exists(&temp_path);
 

--- a/src/tasks/hadolint.rs
+++ b/src/tasks/hadolint.rs
@@ -1,14 +1,15 @@
-use std::env::consts::ARCH;
+use std::env::consts::{ARCH, EXE_SUFFIX};
 
 use inflector::Inflector;
 use regex::Regex;
 
-use crate::lib::{
-    ghrtask::GHRTask,
-    task::{self, Task},
+use crate::{
+    lib::{
+        ghrtask::GHRTask,
+        task::{self, Task},
+    },
+    utils::{github::Asset, golang::os},
 };
-use crate::utils::github::Asset;
-use crate::utils::golang::os;
 
 pub fn task() -> Task {
     Task {
@@ -31,12 +32,7 @@ const GHR_TASK: GHRTask = GHRTask {
 
 fn asset_filter(asset: &Asset) -> bool {
     let os_title = os().to_title_case();
-
-    #[cfg(windows)]
-    let name = format!("hadolint-{}-{}.exe", os_title, ARCH);
-    #[cfg(not(windows))]
-    let name = format!("hadolint-{}-{}", os_title, ARCH);
-
+    let name = format!("hadolint-{}-{}{}", os_title, ARCH, EXE_SUFFIX);
     asset.name == name
 }
 

--- a/src/tasks/jq.rs
+++ b/src/tasks/jq.rs
@@ -1,10 +1,12 @@
-use std::env::consts::{ARCH, OS};
+use std::env::consts::{ARCH, EXE_SUFFIX, OS};
 
-use crate::lib::{
-    ghrtask::GHRTask,
-    task::{self, Task},
+use crate::{
+    lib::{
+        ghrtask::GHRTask,
+        task::{self, Task},
+    },
+    utils::github::Asset,
 };
-use crate::utils::github::Asset;
 
 pub fn task() -> Task {
     Task {
@@ -26,15 +28,11 @@ const GHR_TASK: GHRTask = GHRTask {
 };
 
 fn asset_filter(asset: &Asset) -> bool {
-    #[cfg(windows)]
-    let name = format!("jq-{}.exe", os_arch());
-    #[cfg(not(windows))]
-    let name = format!("jq-{}", os_arch());
-
+    let name = format!("jq-{}{}", os_arch(), EXE_SUFFIX);
     asset.name == name
 }
 
-// this is unfortunately only true for jq 1.5,
+// this is unfortunately only true for jq 1.5 and 1.6,
 // may need to make this smarter to match all past and future versions
 fn os_arch() -> String {
     if ARCH == "x86_64" && OS == "macos" {

--- a/src/tasks/minikube.rs
+++ b/src/tasks/minikube.rs
@@ -1,9 +1,13 @@
-use crate::lib::{
-    ghrtask::GHRTask,
-    task::{self, Task},
+use crate::{
+    lib::{
+        ghrtask::GHRTask,
+        task::{self, Task},
+    },
+    utils::{
+        github::Asset,
+        golang::{arch, os},
+    },
 };
-use crate::utils::github::Asset;
-use crate::utils::golang::{arch, os};
 
 pub fn task() -> Task {
     Task {

--- a/src/tasks/mod.rs
+++ b/src/tasks/mod.rs
@@ -34,7 +34,6 @@ mod tmux;
 mod vale;
 mod vim;
 mod vscode;
-#[cfg(windows)]
 mod windows;
 mod yq;
 mod zsh;
@@ -140,7 +139,6 @@ fn tasks() -> Vec<Task> {
         vscode::task(), // deps: config
         #[cfg(not(windows))]
         zsh::task(), // deps: config
-        #[cfg(windows)]
         windows::task(),
     ]
 }

--- a/src/tasks/nodejs.rs
+++ b/src/tasks/nodejs.rs
@@ -1,4 +1,4 @@
-use std::{self, io, str};
+use std::{self, env::consts::OS, io, str};
 
 use crate::lib::{
     env::Exports,
@@ -37,19 +37,12 @@ where
 
     let prefix = format!("node-{}-{}-{}", version.as_ref(), os(), arch());
 
-    #[cfg(windows)]
     let remote_url = format!(
-        "https://nodejs.org/dist/{}/{}.zip",
+        "https://nodejs.org/dist/{}/{}.{}",
         version.as_ref(),
-        &prefix
+        &prefix,
+        if OS == "windows" { "zip" } else { "tar.gz" },
     );
-    #[cfg(not(windows))]
-    let remote_url = format!(
-        "https://nodejs.org/dist/{}/{}.tar.gz",
-        version.as_ref(),
-        &prefix
-    );
-
     match utils::http::download(remote_url, &temp_path) {
         Ok(()) => {}
         Err(error) => {
@@ -64,10 +57,11 @@ where
     let interim_path = local_path.join(&prefix);
     utils::fs::delete_if_exists(&interim_path);
 
-    #[cfg(windows)]
-    utils::archive::extract_zip(&temp_path, &local_path)?;
-    #[cfg(not(windows))]
-    utils::archive::extract_tar_gz(&temp_path, &local_path)?;
+    if OS == "windows" {
+        utils::archive::extract_zip(&temp_path, &local_path)?;
+    } else {
+        utils::archive::extract_tar_gz(&temp_path, &local_path)?;
+    }
 
     let target_path = local_path.join("node");
     utils::fs::delete_if_exists(&target_path);

--- a/src/tasks/skaffold.rs
+++ b/src/tasks/skaffold.rs
@@ -1,9 +1,15 @@
-use crate::lib::{
-    ghrtask::GHRTask,
-    task::{self, Task},
+use std::env::consts::EXE_SUFFIX;
+
+use crate::{
+    lib::{
+        ghrtask::GHRTask,
+        task::{self, Task},
+    },
+    utils::{
+        github::Asset,
+        golang::{arch, os},
+    },
 };
-use crate::utils::github::Asset;
-use crate::utils::golang::{arch, os};
 
 pub fn task() -> Task {
     Task {
@@ -25,10 +31,7 @@ const GHR_TASK: GHRTask = GHRTask {
 };
 
 fn asset_filter(asset: &Asset) -> bool {
-    #[cfg(windows)]
-    let name = format!("skaffold-{}-{}.exe", os(), arch());
-    #[cfg(not(windows))]
-    let name = format!("skaffold-{}-{}", os(), arch());
+    let name = format!("skaffold-{}-{}{}", os(), arch(), EXE_SUFFIX);
 
     asset.name == name
 }

--- a/src/tasks/vale.rs
+++ b/src/tasks/vale.rs
@@ -36,10 +36,16 @@ pub fn arch() -> &'static str {
 }
 
 fn asset_filter(asset: &Asset) -> bool {
-    #[cfg(windows)]
-    let re = Regex::new(&format!(r"^vale_.*_{}_{}\.zip$", os(), arch())).unwrap();
-    #[cfg(not(windows))]
-    let re = Regex::new(&format!(r"^vale_.*_{}_{}\.tar\.gz$", os(), arch())).unwrap();
+    let re = Regex::new(&format!(
+        r"^vale_.*_{}_{}{}$",
+        os(),
+        arch(),
+        if OS == "windows" {
+            r"\.zip"
+        } else {
+            r"\.tar\.gz"
+        }
+    )).unwrap();
 
     re.is_match(&asset.name)
 }

--- a/src/tasks/vscode.rs
+++ b/src/tasks/vscode.rs
@@ -1,10 +1,11 @@
-use std::path::Path;
-use std::{fs, str};
+use std::{env::consts::OS, fs, path::Path, str};
 
 use toml;
 
-use crate::lib::task::{self, Status, Task};
-use crate::utils;
+use crate::{
+    lib::task::{self, Status, Task},
+    utils,
+};
 
 const COMMAND: &str = "code";
 
@@ -74,12 +75,11 @@ fn sync() -> task::Result {
 
     let src = utils::env::home_dir().join(".dotfiles/config/vscode.json");
 
-    #[cfg(target_os = "macos")]
-    let settings_path = "Library/Application Support/Code/User/settings.json";
-    #[cfg(windows)]
-    let settings_path = "AppData/Roaming/Code/User/settings.json";
-    #[cfg(not(any(target_os = "macos", windows)))]
-    let settings_path = ".config/Code/User/settings.json";
+    let settings_path = match OS {
+        "macos" => "Library/Application Support/Code/User/settings.json",
+        "windows" => "AppData/Roaming/Code/User/settings.json",
+        _ => ".config/Code/User/settings.json",
+    };
     let dest = utils::env::home_dir().join(Path::new(settings_path));
 
     utils::fs::symbolic_link_if_exists(&src, &dest)?;

--- a/src/tasks/windows.rs
+++ b/src/tasks/windows.rs
@@ -1,5 +1,7 @@
-use lib::task::{self, Status, Task};
-use utils;
+use std::env::consts::OS;
+
+use crate::lib::task::{self, Status, Task};
+use crate::utils;
 
 pub fn task() -> Task {
     Task {
@@ -10,6 +12,10 @@ pub fn task() -> Task {
 }
 
 fn sync() -> task::Result {
+    if OS == "windows" {
+        return Ok(Status::Skipped);
+    }
+
     println!("windows: manually configure %PATH% to include:");
 
     let bin_path = utils::env::home_dir().join(".local").join("bin");

--- a/src/tasks/yq.rs
+++ b/src/tasks/yq.rs
@@ -1,9 +1,15 @@
-use crate::lib::{
-    ghrtask::GHRTask,
-    task::{self, Task},
+use std::env::consts::EXE_SUFFIX;
+
+use crate::{
+    lib::{
+        ghrtask::GHRTask,
+        task::{self, Task},
+    },
+    utils::{
+        github::Asset,
+        golang::{arch, os},
+    },
 };
-use crate::utils::github::Asset;
-use crate::utils::golang::{arch, os};
 
 pub fn task() -> Task {
     Task {
@@ -25,10 +31,7 @@ const GHR_TASK: GHRTask = GHRTask {
 };
 
 fn asset_filter(asset: &Asset) -> bool {
-    #[cfg(windows)]
-    let name = format!("yq_{}_{}.exe", os(), arch());
-    #[cfg(not(windows))]
-    let name = format!("yq_{}_{}", os(), arch());
+    let name = format!("yq_{}_{}{}", os(), arch(), EXE_SUFFIX);
 
     asset.name == name
 }

--- a/src/utils/golang.rs
+++ b/src/utils/golang.rs
@@ -20,11 +20,7 @@ pub fn bin_dir() -> PathBuf {
 }
 
 pub fn current_version() -> String {
-    #[cfg(windows)]
-    let exe_path = bin_dir().join("go.exe");
-    #[cfg(not(windows))]
-    let exe_path = bin_dir().join("go");
-
+    let exe_path = bin_dir().join(if OS == "windows" { "go.exe" } else { "go" });
     match utils::process::command_output(exe_path.to_str().unwrap(), &["version"]) {
         Ok(output) => {
             let stdout = str::from_utf8(&output.stdout).unwrap_or_default().trim();
@@ -45,11 +41,7 @@ pub fn goroot() -> PathBuf {
 }
 
 pub fn is_installed() -> bool {
-    #[cfg(windows)]
-    let exe_path = bin_dir().join("go.exe");
-    #[cfg(not(windows))]
-    let exe_path = bin_dir().join("go");
-
+    let exe_path = bin_dir().join(if OS == "windows" { "go.exe" } else { "go" });
     exe_path.is_file()
 }
 

--- a/src/utils/nodejs.rs
+++ b/src/utils/nodejs.rs
@@ -25,10 +25,11 @@ pub fn arch() -> &'static str {
 }
 
 pub fn bin_dir() -> PathBuf {
-    #[cfg(windows)]
-    return install_path();
-    #[cfg(not(windows))]
-    return install_path().join("bin");
+    if OS == "windows" {
+        install_path()
+    } else {
+        install_path().join("bin")
+    }
 }
 
 pub fn current_version() -> String {
@@ -43,12 +44,9 @@ pub fn current_version() -> String {
 }
 
 pub fn has_node() -> bool {
-    #[cfg(windows)]
-    let exe_path = bin_dir().join("node.exe");
-    #[cfg(not(windows))]
-    let exe_path = bin_dir().join("node");
-
-    exe_path.is_file()
+    bin_dir()
+        .join(if OS == "windows" { "node.exe" } else { "node" })
+        .is_file()
 }
 
 pub fn has_npm() -> bool {
@@ -106,10 +104,11 @@ pub fn latest_version() -> String {
 }
 
 pub fn lib_dir() -> PathBuf {
-    #[cfg(windows)]
-    return install_path();
-    #[cfg(not(windows))]
-    return install_path().join("lib");
+    if OS == "windows" {
+        install_path()
+    } else {
+        install_path().join("lib")
+    }
 }
 
 pub fn os() -> &'static str {

--- a/src/utils/process.rs
+++ b/src/utils/process.rs
@@ -1,4 +1,5 @@
 use std::{
+    env::consts::OS,
     ffi::{OsStr, OsString},
     io,
     process::{Command, ExitStatus, Output},
@@ -7,54 +8,40 @@ use std::{
 
 use crate::tasks;
 
-#[cfg(not(windows))]
 #[cfg_attr(feature = "cargo-clippy", allow(needless_pass_by_value))]
 pub fn command_output<O, S>(cmd: O, args: &[S]) -> io::Result<Output>
 where
     O: Into<OsString> + AsRef<OsStr>,
     S: Into<String> + AsRef<str>,
 {
-    let cmd_args: Vec<&str> = args.into_iter().map(|s| s.as_ref()).collect();
-    with_env(Command::new(cmd).args(cmd_args)).output()
+    if OS == "windows" {
+        let cmd_os = cmd.into();
+        let cmd_lossy = cmd_os.to_string_lossy();
+        let mut cmd_args: Vec<&str> = vec!["/c", cmd_lossy.as_ref()];
+        cmd_args.extend::<Vec<&str>>(args.into_iter().map(|s| s.as_ref()).collect());
+        with_env(Command::new("cmd").args(cmd_args)).output()
+    } else {
+        let cmd_args: Vec<&str> = args.into_iter().map(|s| s.as_ref()).collect();
+        with_env(Command::new(cmd).args(cmd_args)).output()
+    }
 }
 
-#[cfg(windows)]
-#[cfg_attr(feature = "cargo-clippy", allow(needless_pass_by_value))]
-pub fn command_output<O, S>(cmd: O, args: &[S]) -> io::Result<Output>
-where
-    O: Into<OsString> + AsRef<OsStr>,
-    S: Into<String> + AsRef<str>,
-{
-    let cmd_os = cmd.into();
-    let cmd_lossy = cmd_os.to_string_lossy();
-    let mut cmd_args: Vec<&str> = vec!["/c", cmd_lossy.as_ref()];
-    cmd_args.extend::<Vec<&str>>(args.into_iter().map(|s| s.as_ref()).collect());
-    with_env(Command::new("cmd").args(cmd_args)).output()
-}
-
-#[cfg(not(windows))]
 #[cfg_attr(feature = "cargo-clippy", allow(needless_pass_by_value))]
 pub fn command_spawn_wait<O, S>(cmd: O, args: &[S]) -> io::Result<ExitStatus>
 where
     O: Into<OsString> + AsRef<OsStr>,
     S: Into<String> + AsRef<str>,
 {
-    let cmd_args: Vec<&str> = args.into_iter().map(|s| s.as_ref()).collect();
-    with_env(Command::new(cmd).args(cmd_args)).spawn()?.wait()
-}
-
-#[cfg(windows)]
-#[cfg_attr(feature = "cargo-clippy", allow(needless_pass_by_value))]
-pub fn command_spawn_wait<O, S>(cmd: O, args: &[S]) -> io::Result<ExitStatus>
-where
-    O: Into<OsString> + AsRef<OsStr>,
-    S: Into<String> + AsRef<str>,
-{
-    let cmd_os = cmd.into();
-    let cmd_lossy = cmd_os.to_string_lossy();
-    let mut cmd_args: Vec<&str> = vec!["/c", cmd_lossy.as_ref()];
-    cmd_args.extend::<Vec<&str>>(args.into_iter().map(|s| s.as_ref()).collect());
-    with_env(Command::new("cmd").args(cmd_args)).spawn()?.wait()
+    if OS == "windows" {
+        let cmd_os = cmd.into();
+        let cmd_lossy = cmd_os.to_string_lossy();
+        let mut cmd_args: Vec<&str> = vec!["/c", cmd_lossy.as_ref()];
+        cmd_args.extend::<Vec<&str>>(args.into_iter().map(|s| s.as_ref()).collect());
+        with_env(Command::new("cmd").args(cmd_args)).spawn()?.wait()
+    } else {
+        let cmd_args: Vec<&str> = args.into_iter().map(|s| s.as_ref()).collect();
+        with_env(Command::new(cmd).args(cmd_args)).spawn()?.wait()
+    }
 }
 
 fn with_env(c: &mut Command) -> &mut Command {
@@ -99,20 +86,19 @@ mod tests {
     #[test]
     fn command_spawn_wait_does_not_exist() {
         match command_spawn_wait("does_not_exist", &["nope"]) {
-            #[cfg(not(windows))]
-            Ok(_status) => {
-                assert!(false);
-            }
-            #[cfg(windows)]
             Ok(status) => {
-                assert!(!status.success());
+                if OS == "windows" {
+                    assert!(!status.success());
+                } else {
+                    assert!(false);
+                }
             }
             Err(_error) => {
-                #[cfg(not(windows))]
-                assert!(true);
-
-                #[cfg(windows)]
-                assert!(false);
+                if OS == "windows" {
+                    assert!(false);
+                } else {
+                    assert!(true);
+                }
             }
         }
     }

--- a/src/utils/ssh.rs
+++ b/src/utils/ssh.rs
@@ -879,8 +879,7 @@ where
 
 #[cfg(test)]
 mod tests {
-    use std::fs;
-    use std::path::Path;
+    use std::{env::consts::OS, fs, path::Path};
 
     use super::*;
 
@@ -946,13 +945,13 @@ mod tests {
         let config_path =
             Path::new(env!("CARGO_MANIFEST_DIR")).join("tests/fixtures/ssh_config.output.txt");
 
-        #[cfg(windows)]
-        let want = fs::read_to_string(&config_path)
-            .unwrap()
-            .replace("IdentityFile ~/.ssh/id_rsa", "IdentityFile ~\\.ssh\\id_rsa");
-        #[cfg(not(windows))]
-        let want = fs::read_to_string(&config_path).unwrap();
-
+        let want = if OS == "windows" {
+            fs::read_to_string(&config_path)
+                .unwrap()
+                .replace("IdentityFile ~/.ssh/id_rsa", "IdentityFile ~\\.ssh\\id_rsa")
+        } else {
+            fs::read_to_string(&config_path).unwrap()
+        };
         let config = example_config();
 
         assert_eq!(String::from(&config), want);

